### PR TITLE
feat: js dom, js async, nodejs

### PR DESCRIPTION
**Pull request by mountainmort**
This PR fulfills the standards in JS DOM Manipulation #7 :
- [x] Select the <body> element
- [x] Add a list element as a child of the body
- [x] Add 2 button as children to the <body> element
- [x] The first button should add an list item element to the newly created list
- [x] The second button should remove a list item from the list
- [x] Make an event that listens for a click on one of the list items and when that event happens something updates (example: the background color updates)

And with the second commit, it also fulfills requirements for Async Javascript #8:
- [x] Create an HTML button that on click that uses a JS Promise to wait one second and then sends a fetch request to GET data from an external API
- [x] Once the request is sent show that the request is loading
- [x] If the request succeeds show the contents of the received data to the webpage
- [x] If the request fails show the error status code and message

To fire off the API Request for #8, click on this button at the bottom of the page. It will dynamically add a new section to the page and populate a table with the data:
image

Additional commit added here, fulfills requirements for Nodejs #9. To run the local node code, you'll need to pull this local and run `yarn nodetest` in the console (unless there's a way to access the console via Netlify?).
- [x] Delete the exported add function
- [x] Define 4 functions
- [x] 1 function that given a file path creates a file on the file system and logs out whether the file was created or not
- [x] 1 function that given a file path reads a file on the file system and logs the contents out
- [x] 1 function that given a file path deletes a file on the file system and logs out whether it successfully deleted that file or not
- [x] 1 function that given a file path and a string updates the file at the file path with the given string and logs out whether it was successful
- [x] After the function definitions create a variable and assign it a string that is a file path using process.cwd and path.join. Call the functions in the below order passing it in as a parameter.
- [x] Call the create file function
- [x] Call the Update file function with a random string
- [x] Call the Read file function
- [x] Call the Delete file function

Quick note: to get the deploy preview to work, I had to make a few minor tweaks to the Netlify deploy settings. 
image

### CODE REVIEW

File: src/resources/js/main.js [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+    let dynamicDiv = document.getElementById('dynamic-content');
+    appendButton(dynamicDiv, 'Add List Item', 'button-add', 'button-add dynamic-list-button', () => { modifyList('dynamic-list', 'dynamic-list-li', 'add'); });
+    appendButton(dynamicDiv, 'Remove List Item', 'button-remove', 'button-remove dynamic-list-button', () => { modifyList('dynamic-list', 'dynamic-list-li', 'remove') });
+    
+    // #region Helper Functions
+    
+    function createStyles() {
+        let style = document.createElement('style');
+        style.type = 'text/css';
+        style.innerHTML = `
```
Comment (axecopfire) [inline src/resources/js/main.js:123]: This works, not how I expected, but it's good. I was looking to use `addClassList`/`removeClassList` or also `<element>.style = {}` I think I can be more explicit. I didn't realize you could do what you're doing, which is cool.

**Comment by axecopfire** [comment]
I'll add a TODO for having a copy pasta Github hook that could run `npm run test` and show the results in PRs. I was thinking that it could be cool to just have a single VWC Github action that we'd add and say "Just add this <script> to your package.json" and the action would run it and show the output in the PR

**Review by axecopfire — APPROVED** [review]

File: src/resources/js/main.js (hunk 1 above)
Comment (mountainmort) [reply src/resources/js/main.js:123]: Understand what you were going for here after our talk and will review the `addClassList` / `removeClassList` APIs as a follow up.

**Comment by mountainmort** [comment]
> I'll add a TODO for having a copy pasta Github hook that could run `npm run test` and show the results in PRs. I was thinking that it could be cool to just have a single VWC Github action that we'd add and say "Just add this <script> to your package.json" and the action would run it and show the output in the PR

This makes sense. If it was predefined what script action the new code should be tied to, then it could be automated.

**Pull request merged**